src/ui_session_interface.rs

```diff
@@ -216,6 +216,11 @@ impl<T: InvokeUiSession> Session<T> {
 
     pub fn get_keyboard_mode(&self) -> String {
         let mode = self.lc.read().unwrap().keyboard_mode.clone();
+        if self.peer_platform() == crate::PLATFORM_ANDROID {
```

Comment:
👍 

This commit seems better than
https://github.com/rustdesk/rustdesk/pull/6168/

It's independent of the UI.

Reply:
But the logic here is the same as in flutter_ffi.rs. Maybe it's better to merge the code.
```rust
if session.peer_platform() == "Android" && mode == KeyboardMode::Map {
    return SyncReturn(false);
}
```

Reply:
The logic in flutter_ffi.rs is used to hide map mode in UI, ui_session_interface.rs is used to override keyboard mode saved in local storage. It’s two part of the same problem.

Reply:
> It’s two part of the same problem.

Yes, but if something changed, you have to change to two places, which is error prone.

They are doing almost the same filtering, maybe you can extract the filter code, or some func like `should_replace_mode_to(conds, mode) -> Option<mode>`

Reply:
I agree, that's better. Working on it.

Reply:
Done.